DllImportGenerator/designs/StructMarshalling.md

```diff
@@ -61,6 +61,27 @@ The analyzer will report an error if neither the construtor nor the ToManaged me
 
 If the native type `TNative` also has a public `Value` property, then the value of the `Value` property will be passed to native code instead of the `TNative` value itself. As a result, the type `TNative` will be allowed to be non-blittable and the type of the `Value` property will be required to be blittable. If the `Value` property is settable, then when marshalling in the native-to-managed direction, a default value of `TNative` will have its `Value` property set to the native value. If `Value` does not have a setter, then marshalling from native to managed is not supported.
 
+If the `Value` property has a `ref` or `readonly ref` return type, then the marshalling type is only supported in scenarios where pinning is available. See below for an example of a managed type with a native type with a `Value` property.
```

Comment:
Yes that's currently the case. I'm not a fan of this, so I think I'm going to try an alternative approach where instead of supporting a ref return `Value` property, we support a `GetPinnableReference()` method on the marshaller type. That should remove this restriction and make it possible to support all the scenarios we want with one marshaller type.